theano/tensor/elemwise.py

```diff
@@ -849,6 +849,14 @@ def prepare_node(self, node, storage_map, compute_map):
             char = numpy.sctype2char(out_dtype)
             sig = char * node.nin + '->' + char * node.nout
             node.tag.sig = sig
+        node.tag.fake_node = Apply(
+            self.scalar_op,
+            [get_scalar_type(dtype=input.type.dtype).make_variable()
+             for input in node.inputs],
+            [get_scalar_type(dtype=output.type.dtype).make_variable()
+             for output in node.outputs])
```

Comment:
This is a fake node. This is faster then calling make_node as it don't do many checks. I don't want to run the risk of slowing that down compared to the old code. I took it from line 1112 in the old version.